### ISSUE TITLE
Impossible pass empty list of non-string type as query param for route with non-empty default value

### ISSUE THREAD

**Issue by ihostage** [playframework/playframework#12979]
For example we have a next routes in our tests:
```
GET         /take-slist-int-d               controllers.Application.takeListIntWithDefault(x: List[Int] ?= List(1, 2, 3))
```
And for this route we can't to pass an empty list as value of param `x`, because current behavior for all URL variants (`/take-slist-int-d?x`, `/take-slist-int-d?x=`  and `/take-slist-int-d`) is similar and delegating to default value.

At the same moment the behavior for list of string param is different and pass an empty list is possible
```
GET         /take-slist-str-d               controllers.Application.takeListStringWithDefault(x: List[String] ?= List("abc", "def", "ghi"))
```

This difference easy to see in tests

https://github.com/playframework/playframework/blob/77fb9a1acd9af52ad11b78038d4bce112a485fc9/dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/routes-compiler-routes-compilation/tests/RouterSpec.scala#L727-L734

https://github.com/playframework/playframework/blob/77fb9a1acd9af52ad11b78038d4bce112a485fc9/dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/routes-compiler-routes-compilation/tests/RouterSpec.scala#L639-L646

I find this difference confusable 🤔

@mkurz WDYT?

**Comment by ihostage** [comment]
Oh, no, I made mistake. For `List[String]` we also can't to pass an empty list, because `?x=` or `?x` will be interpreted as list with one empty string element. 🤔

**Comment by mkurz** [comment]
Is this no still a problem or was this fixed with your PRs ?

**Comment by ihostage** [comment]
Nope, it is. But I still haven't found a good way to solve this very rare case 🤷‍♂

**Comment by mkurz** [comment]
Did some testing to better understand what's going on:

|  | `?x=` | `?x` | `?` | `?x=foo` |
|--|-------|------|-----|----------|
| `/list-str` | 1 \[\] | 1 \[\] | 3 \[abc, def, ghi\]✅ | 1 \[foo\] ✅ |
| `/list-int` | 3 \[1, 2, 3\] | 3 \[1, 2, 3\] | 3 \[1, 2, 3\]✅ | 400 bad request ✅ |
| `/str` | \<empty-string\> | \<empty-string\> | abc✅ | foo ✅ |
| `/int` | 123 | 123 | 123✅ | 400 bad request ✅ |
| `/opt-str` | Optional\[\] | Optional\[\] | Optional\[abc\]✅ | Optional\[foo\] ✅ |
| `/opt-int` | Optional\[123\] | Optional\[123\] | Optional\[123\]✅ | 400 bad request ✅ |

<sub>1 [] = array with size 1 (containing empty element)</sub>
<sub>Optional[] = Non-empty optional with empty string (otherwise we would see Optional.empty)</sub>

- `?`: Cases where the param is not passed is ok, because then the default is used.
- `?x=foo`: Cases where a value is passed ("foo") is ok, because for string it will use the value and for int it will fail with http 400.

Thinking about the other cases... Like what if a user wants to explicitly pass an 
- empty string?
- empty list (=size 0)?
- list with with one element (works for empty string only - there is no empty primitive like int)?
- Optional.empty?

**Comment by ihostage** [comment]
Awesome visualization of current behavior, Matthias! 👍 
Personally, I think if we will recognize a `?x=` and `?x` as not the same case, it will give us a flexibility to cover more use cases.
For example for routes with default value the target behaviour can be next:
|  | `?x=` | `?x` | `?` |
|--|-------|------|------|
| `/list-str` | \[""\] | \[\] | ["abc", "def", "ghi"] |
| `/list-int` | 400 bad request | \[\] | [1, 2, 3] |
